src/exporters/prometheus.rs

```diff
@@ -170,71 +226,135 @@ fn push_metric(
 }
 
 /// Handles requests and returns data formated for Prometheus.
-async fn show_metrics(data: web::Data<PowerMetrics>) -> impl Responder {
-    let now = current_system_time_since_epoch();
-    let mut last_request = data.last_request.lock().unwrap();
-
-    if now - (*last_request) > Duration::from_secs(5) {
-        {
-            info!(
-                "{}: Refresh topology",
-                Utc::now().format("%Y-%m-%dT%H:%M:%S")
-            );
-            let mut topology = data.topology.lock().unwrap();
-            (*topology)
-                .proc_tracker
-                .clean_terminated_process_records_vectors();
-            (*topology).refresh();
+async fn show_metrics(
+    req: Request<Body>,
+    context: Arc<PowerMetrics>,
+    suffix: String,
+) -> Result<Response<Body>, Infallible> {
+    warn!("{}", req.uri());
+    let mut body = String::new();
+    if req.uri().path() == format!("/{}", &suffix) {
+        warn!("in metrics !");
+        let now = current_system_time_since_epoch();
+        let mut last_request = context.last_request.lock().unwrap();
+        if now - (*last_request) > Duration::from_secs(5) {
+            {
+                info!(
+                    "{}: Refresh topology",
+                    Utc::now().format("%Y-%m-%dT%H:%M:%S")
```

Comment:
FYI, I will make a PR to change the logging stuff. The way it is done is not really good.

Reply:
What is the part that worries you ?